test/unit/logic/account.js

```
@@ -0,0 +1,291 @@
+var sinon = require('sinon');
+
+var chai = require('chai');
+var expect = require('chai').expect;
+var _  = require('lodash');
+
+var AccountLogic = require('../../../logic/account.js');
+var modulesLoader = require('../../common/initModule').modulesLoader;
+
+var validAccount = {
+	username: 'genesis_100',
+	isDelegate: 1,
+	u_isDelegate: 1,
+	secondSignature: 0,
+	u_secondSignature: 0,
+	u_username: 'genesis_100',
+	address: '10881167371402274308L',
+	publicKey: 'addb0e15a44b0fdc6ff291be28d8c98f5551d0cd9218d749e30ddb87c6e31ca9',
+	secondPublicKey: null,
+	balance: '231386135',
+	u_balance: '231386135',
+	vote: '9820020609280331',
+	rate: '0',
+	delegates: null,
+	u_delegates: null,
+	multisignatures: null,
+	u_multisignatures: null,
+	multimin: 0,
+	u_multimin: 0,
+	multilifetime: 0,
+	u_multilifetime: 0,
+	blockId: '10352824351134264746',
+	nameexist: 0,
+	u_nameexist: 0,
+	producedblocks: 27,
+	missedblocks: 1,
+	fees: '231386135',
+	rewards: '0',
+	virgin: 1
+};
+
+describe('account', function () {
+
+	var account; 
+
+	before(function (done) {
+		modulesLoader.initLogicWithDb(AccountLogic, function (err, __account) {
+			expect(err).to.not.exist;
+			account = __account;
+			done();
+		}, {});
+	});
+	describe('Account', function () {
+
+	});
+	describe('createTables', function () {
+
+	});
+	describe('removeTables', function () {
+
+	});
+	describe('objectNormalize', function () {
+		it.skip('should be okay for a valid account object', function () {
+			expect(account.objectNormalize(validAccount)).to.be.an('object');
+		});
+	});
+
+	describe('verifyPublicKey', function () {
+		it('should be okay for empty params', function () {
+			expect(account.verifyPublicKey()).to.be.undefined;
+		});
+
+		it('should throw if parameter is not a string', function () {
+			expect(function () {
+				account.verifyPublicKey(1);
+			}).to.throw('Invalid public key, must be a string');
+		});
+
+		it('should throw if parameter is of wrong size', function () {
```

should throw if parameter is an invalid length